mubench.pipeline/requirements.py

```diff
@@ -74,3 +77,56 @@ def __init__(self):
 
     def check(self):
         pass
+
+class CPUCountRequirement(Requirement):
+    MIN_CPU_COUNT = 2
+
+    def __init__(self):
+        super().__init__("CPUs >= {}".format(self.MIN_CPU_COUNT))
+
+    def check(self):
+        cpu_count = self._get_cpu_count()
+        if cpu_count < self.MIN_CPU_COUNT:
+            raise ValueError("Only {} CPUs available.".format(cpu_count))
+
+    def _get_cpu_count(self):
+        if _in_container():
+            return self._get_container_cpu_count()
+        else:
+            return psutil.cpu_count(logical=False)
```

Comment:
Anyways, since these dependencies are actually requirements themselves now... how about we add a fallback in case people don't have them? For the CPUs I would just issue a warning stating that we couldn't determine the CPI count. For the memory, we could default to outputting the unformatted size.